packages/web3-shared/src/hooks/context/shared-states.ts

```diff
@@ -0,0 +1,37 @@
+import { ChainId } from '..'
+import { useWeb3Provider } from './provider'
+import { useSubscription } from 'use-subscription'
+import type { Wallet } from '../useWallets'
+
+/** @internal */
+export function useSharedStatesRaw() {
+    const context = useWeb3Provider()
+
+    const wallets = useSubscription(context.wallets)
+    const selectedAddress = useSubscription(context.selectedWalletAddress)
+    const chainID = useChainID()
+    const isChainIDAvailable = useChainIDAvailable(wallets)
+    return {
+        chainID,
+        isChainIDAvailable,
+        wallets,
+        selectedAddress,
+    }
+}
+
+/** @internal */
+function useChainID(): ChainId {
+    const _ = useWeb3Provider()
+    const allowTestChain = useSubscription(_.allowTestChain)
+    const chain = useSubscription(_.currentChain)
+    if (!allowTestChain) return ChainId.Mainnet
+    return chain
+}
+/** @internal */
+function useChainIDAvailable(wallets: Wallet[]): boolean {
+    const _ = useWeb3Provider()
+    const allowTestChain = useSubscription(_.allowTestChain)
+    const unsafeChainId = useSubscription(_.currentChain)
+    const selectedWallet = wallets
```

Comment:
It looks like a mistake

Reply:
The `wallets` look not be **selected** wallet

Reply:
fixed